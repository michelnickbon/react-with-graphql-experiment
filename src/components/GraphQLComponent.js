import React from 'react';
import GraphQLManager from '../assets/GraphQLManager';
import { Button } from '@material-ui/core';

class GraphQLComponent extends React.Component {

    constructor() {
        super();
        this.state = {
          result: "Press the button to view the result",
          query: `query { Lift(id: "panorama") { name status }}`
        };
        this.launchQuery = this.launchQuery.bind(this)
    }

    // Launch GraphQL query
    async launchQuery() {
      this.setState({ result: "Loading..." });
      const result = await GraphQLManager.GetGraphQLData(this.state.query);
      this.setState({ result: JSON.stringify(result, null, 2) });
    }

    render() {
        return (
          <div className="container">
            <div className="intro"> 
              <h1>Hello there :-)</h1>
              <p> - Lets try some GraphQL! </p>
            </div>
            <div className="button">
              <Button onClick={this.launchQuery} variant="contained" color="primary" disableElevation> Run GraphQL query </Button>
            </div>
            <div className="query">
              <p> {this.state.query} </p>
            </div>
            <div className="result">
              {this.state.result}
            </div>
          </div>
        );
    }
}

export default GraphQLComponent;