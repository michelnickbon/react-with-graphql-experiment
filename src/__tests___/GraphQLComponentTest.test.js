import React from 'react';
import renderer from 'react-test-renderer';
import GraphQLComponent from '../components/GraphQLComponent';

test('Snapshot is rendered', () => {
    const component = renderer.create(<GraphQLComponent />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});