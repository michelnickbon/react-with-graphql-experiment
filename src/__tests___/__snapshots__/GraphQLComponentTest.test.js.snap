// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot is rendered 1`] = `
<div
  className="container"
>
  <div
    className="intro"
  >
    <h1>
      Hello there :-)
    </h1>
    <p>
       - Lets try some GraphQL! 
    </p>
  </div>
  <div
    className="button"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-disableElevation"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
         Run GraphQL query 
      </span>
    </button>
  </div>
  <div
    className="query"
  >
    <p>
       
      query {
            Country(alpha2Code: "SE") {
              name
              nativeName
              population
              convertedArea(areaUnit: SQUARE_MILES) {
                populationDensity
              }
              capital
              location {
                latitude
                longitude
              }
              currencies {
                name
                symbol
              }
              flag {
                emoji
                svgFile
              }
            }
          }
       
    </p>
  </div>
  <div
    className="result"
  >
    Press the button to view the result
  </div>
</div>
`;
